### PR TITLE
TTL

### PR DESCRIPTION
**Pull request by faze-quadrata**
## Checklist
1. [ ] Unit Testing?
2. [ ] Backward Compatible?
3. [ ] Upgrade Safe (storage)?
4. [ ] Updated Confluence?
5. [ ] Updated Gitbook?

### CODE REVIEW

File: contracts/QuadPassport.sol [hunk 1]
```diff
@@ -109,8 +109,15 @@ contract QuadPassport is IQuadPassport, UUPSUpgradeable, QuadSoulbound, QuadPass
     ) internal returns(address) {
         require(msg.value == _config.fee,  "INVALID_SET_ATTRIBUTE_FEE");
         require(governance.eligibleTokenId(_config.tokenId), "PASSPORT_TOKENID_INVALID");
+
+        require(_config.verifiedAt != 0, "VERIFIED_AT_CANNOT_BE_ZERO");
         require(_config.issuedAt != 0, "ISSUED_AT_CANNOT_BE_ZERO");
-        require(_config.issuedAt <= block.timestamp, "INVALID_ISSUED_AT");
+
+        require(_config.verifiedAt <= block.timestamp, "FUTURE_VERIFIED_AT");
+        require(_config.issuedAt <= block.timestamp, "FUTURE_ISSUED_AT");
```
Comment (fabrice-quadrata) [inline contracts/QuadPassport.sol:117]: I think that's technically possible to be a bit in the future for issuedAt?

File: contracts/QuadPassport.sol [hunk 2]
```diff
@@ -119,6 +126,7 @@ contract QuadPassport is IQuadPassport, UUPSUpgradeable, QuadSoulbound, QuadPass
                 _config.attrKeys,
                 _config.attrValues,
                 _config.issuedAt,
+                _config.verifiedAt,
```
Comment (fabrice-quadrata) [inline contracts/QuadPassport.sol:129]: can we put verifiedAt before issuedAt?

File: contracts/QuadPassport.sol [hunk 3]
```diff
@@ -109,8 +109,15 @@ contract QuadPassport is IQuadPassport, UUPSUpgradeable, QuadSoulbound, QuadPass
     ) internal returns(address) {
         require(msg.value == _config.fee,  "INVALID_SET_ATTRIBUTE_FEE");
         require(governance.eligibleTokenId(_config.tokenId), "PASSPORT_TOKENID_INVALID");
+
+        require(_config.verifiedAt != 0, "VERIFIED_AT_CANNOT_BE_ZERO");
         require(_config.issuedAt != 0, "ISSUED_AT_CANNOT_BE_ZERO");
-        require(_config.issuedAt <= block.timestamp, "INVALID_ISSUED_AT");
+
+        require(_config.verifiedAt <= block.timestamp, "FUTURE_VERIFIED_AT");
```
Comment (fabrice-quadrata) [inline contracts/QuadPassport.sol:116]:
```suggestion
        require(_config.verifiedAt <= block.timestamp, "INVALID_VERIFIED_AT");
```

File: contracts/QuadPassport.sol [hunk 4]
```diff
@@ -109,8 +109,15 @@ contract QuadPassport is IQuadPassport, UUPSUpgradeable, QuadSoulbound, QuadPass
...
+        require(_config.issuedAt <= block.timestamp, "FUTURE_ISSUED_AT");
+
+        require(block.timestamp <= _config.issuedAt + 1.weeks, "EXPIRED_ISSUED_AT");
```
Comment (fabrice-quadrata) [inline contracts/QuadPassport.sol:119]:
```suggestion
        require(block.timestamp <= _config.issuedAt + 1.days, "EXPIRED_ISSUED_AT");
```

File: contracts/QuadPassport.sol (hunk 1 above)
Comment (faze-quadrata) [reply contracts/QuadPassport.sol:117]: yeah, i can remove this line. right now, issuedAt is practically verifiedAt on SL side

File: contracts/QuadPassport.sol (hunk 2 above)
Comment (faze-quadrata) [reply contracts/QuadPassport.sol:128]: done

**Pull request merged**